crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -409,33 +403,52 @@ impl Format<PyFormatContext<'_>> for FormatTrailingEndOfLineComment<'_> {
 ///   unnecessary allocations.
 /// * If the content is modified then make as few allocations as possible and use
 ///   a dynamic text element at the original slice's start position.
-pub(crate) const fn format_normalized_comment(
-    comment: Cow<'_, str>,
-    range: TextRange,
-) -> FormatNormalizedComment<'_> {
-    FormatNormalizedComment { comment, range }
+const fn format_normalized_comment(comment: NormalizedComment) -> FormatNormalizedComment {
+    FormatNormalizedComment { comment }
 }
 
-pub(crate) struct FormatNormalizedComment<'a> {
-    comment: Cow<'a, str>,
-    range: TextRange,
+struct FormatNormalizedComment {
+    comment: NormalizedComment,
 }
 
-impl Format<PyFormatContext<'_>> for FormatNormalizedComment<'_> {
+impl Format<PyFormatContext<'_>> for FormatNormalizedComment {
```

Comment:
You can implement `Format` directly on `NormalizedComment`. The orphan rule doesn't apply here